### PR TITLE
chore: fix npm command calls from maven

### LINKED ISSUE #168: Random build failure
**Describe the bug**
The full `mvn clean install` on the project randomly ends in failure due to a NPM error on `ara-client` (using the Maven plugin). But the classic npm build on the client pass.

**To Reproduce**
1. Open a terminal and go to the ara root folder
2. Launch `mvn clean install`
3. Randomly the build failed on `ara-client` due to a NPM error
4. Launch `npm build` in `ara/client`, the build pass in success

**Expected behavior**
The `ara-client` maven module should not fail if the host `npm` doesn't.
**Environment**
 - OS: macOS, Windows 10
 - Browser: N/A

### PR DESCRIPTION
**Pull request by omar-chahbouni-decathlon**
## Type of modification

* [ ] Breaking change
* [ ] New Feature
* [ ] Bug Fix
* [X] Chore (refactor, documentation, tests... all the changes with no impact on ARA functionalities.) 

## Changes description

**Add some quick description about what your PullRequest change in
ARA's feature/behavior**

Close #168 

## Technical description

**Add a quick description about what your PullRequest change in 
ARA's code. (example: add a processor check in...)**

## PR CheckList

Please make sure your PullRequest respect all those items :

* [ ] Your PR's title has the prefix : `feat:`, `fix:` or `chore:`
* [ ] Unless your PR is related to a Hotfix (and approved by an ARA maintainer), the targeted branch for it is the branch `version/X.Y.Z` and **not** `master`
* [ ] You have asked a review from one of the ARA maintainer in your PR.
* [ ] If your PR is related to an issue, add the issue's number in it.
* [ ] All the code you added is documented.
* [ ] All the code you added is tested and the  tests are in success.
* [ ] You already signed the [Contributor License Agreement](https://github.com/Decathlon/ara/blob/master/contributor-licence-agreement.adoc) and give us the document

### CODE REVIEW

**Comment by omar-chahbouni-decathlon** [comment]
- [ ] Check what is currently generated and where it is sent ( when launching `mvn clean install`)
- [ ] Write a JS script that does the same thing
- [ ] Add the command to the package.json, in order to later be able to run it (via `npm run`)
- [ ] You should now be able to build the project: delete the generated files (targets, node_modules, etc.) and run `mvn clean install` then check that everything is built correctly
- [ ] Update the CircleCI configuration file to build the client (replacing what maven is currently doing)
- [ ] Getting rid of the pom.xml (client)

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,6 +6,9 @@ executors:
       SERVER_IMAGE_NAME: decathlon/ara-server
     docker:
       - image: circleci/openjdk:8u222-stretch
+  client-builder:
+    docker:
+      - image: circleci/node:6.10
```
Comment (dktsni) [inline .circleci/config.yml]: I think we can take the oppportunity of this work to upgrade the node version used (I managed to work on all the 4.1.0 version with a node 10 locally without any issues)

Maybe using `circleci/node:10` ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -28,6 +31,18 @@ jobs:
             git tag -a v$(cat version.txt) -m "Release version $(cat version.txt)"
             git push --tags origin
             git push origin master
+  build_client:
+    executor: client-builder
+    working_directory: ~/ara/client
+    steps:
+      - checkout:
+          path: ~/ara
```
Comment (dktsni) [inline .circleci/config.yml]: This step seems weird to me. If I understand it : 
- We're checking out all the project inside the path `~/ara` but we're working only in `~/ara/client` 
amiright ?

If so, maybe we should remove the checkout step from this job and the `build_app` one to put it in its own job. Doing so will allow use to only checkout once when we'll use them both (like in `build-pr` and `build-master`. What's your opinion on this ?

File: .circleci/config.yml [hunk 3]
```diff
@@ -28,6 +31,18 @@ jobs:
...
+      - run:
+          name: Prepare client
```
Comment (dktsni) [inline .circleci/config.yml]: `Install client's dependencies` seems more accurate to me 🤔

File: .circleci/config.yml (hunk 1 above)
Comment (omar-chahbouni-decathlon) [reply .circleci/config.yml]: When I used the latest version, I had some issues with `npm install`, so I set the version I currently have on my machine.
I can nonetheless try with this version and see if it works :)

File: .circleci/config.yml (hunk 2 above)
Comment (omar-chahbouni-decathlon) [reply .circleci/config.yml]: I had some trouble reaching the client directory: even though setting working_directory as ~/ara/client, without the checkout, I wasn't able to access it.
This helped resolve the issue -> https://stackoverflow.com/questions/50570221/circleci-2-0-working-with-subdirectory

File: .circleci/config.yml (hunk 3 above)
Comment (omar-chahbouni-decathlon) [reply .circleci/config.yml]: I agree :)

File: .circleci/config.yml (hunk 2 above)
Comment (dktsni) [reply .circleci/config.yml]: Ok for me, but could you has the reason and the link in comment in the YML file please ?

Comment (omar-chahbouni-decathlon) [reply .circleci/config.yml]: done

**Pull request merged**